docs/index.md

```diff
@@ -0,0 +1,233 @@
+# SOCI Index
```

Comment:
nit: Most of our docs contain a TOC at the top. Can you generate one using `doctoc .`?

Reply:
Added. PTAL.